sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcSQLQueryBuilder.scala

```diff
@@ -81,17 +81,21 @@ class JdbcSQLQueryBuilder(dialect: JdbcDialect, options: JDBCOptions) {
   /**
    * Constructs the WHERE clause that following dialect's SQL syntax.
    */
-  def withPredicates(predicates: Array[Predicate], part: JDBCPartition): JdbcSQLQueryBuilder = {
+  def withPredicates(
+      predicates: Array[Predicate],
+      part: Option[JDBCPartition]): JdbcSQLQueryBuilder = {
```

Comment:
This is a public API and we can't change its signature. How about we pass a fake `JDBCPartition` in EXPLAIN?

Reply:
We can do that, we can pass JDBCPartition(whereClause = null, idx = 0).
Or maybe, we can skip withPredicates method from generateJDBCQuery (maybe a little bit nicer.